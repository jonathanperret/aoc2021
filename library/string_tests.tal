~library/macros.tal

( devices )

|00 @System     [ &vector $2 &wst  $1 &rst    $1 &pad   $4 &r $2 &g $2 &b $2   &debug $1 &halt $1 ]
|10 @Console    [ &vector $2 &read $1 &pad    $5 &write $1 &error  $1 ]

( variables )

|0000

|0100 @program

  S< "using 20 "the 20 "S< 20 "macro >S PRINT LF
  ( EXPECT using the S< macro )

  P< "or 20 "the 20 "P< 20 "macro 0a >P
  ( EXPECT or the P< macro )

  BRK !

~library/console.lib.tal
~library/string.tal

@x 10 20 00 00 00 00 00 00
