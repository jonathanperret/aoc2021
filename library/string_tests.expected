using the S< macro
or the P< macro
